sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/common/SchemaManager.java

```diff
@@ -121,6 +145,63 @@ public Uni<Schema.CqlKeyspaceDescribe> getKeyspaceAuthorized(String keyspace) {
             });
   }
 
+  /**
+   * Get all keyspace from the bridge. Prior to getting each keyspace it will execute the schema
+   * authorization request (single request for all available keyspace).
+   *
+   * @return Multi containing Schema.CqlKeyspaceDescribe
+   */
+  @WithSpan
+  public Multi<Schema.CqlKeyspaceDescribe> getKeyspacesAuthorized() {
+    StargateBridge bridge = requestInfo.getStargateBridge();
+
+    // get all keyspace names
+    return getKeyspaceNames(bridge)
+
+        // collect list
+        .collect()
+        .asList()
+
+        // then go for the schema read
+        .onItem()
+        .transformToMulti(
+            keyspaceNames -> {
+
+              // create schema reads for all keyspaces
+              List<Schema.SchemaRead> reads =
+                  keyspaceNames.stream()
+                      .map(n -> SchemaReads.keyspace(n, sourceApi))
+                      .collect(Collectors.toList());
+
+              Schema.AuthorizeSchemaReadsRequest request =
+                  Schema.AuthorizeSchemaReadsRequest.newBuilder().addAllSchemaReads(reads).build();
+
+              // execute request
+              return bridge
+                  .authorizeSchemaReads(request)
+
+                  // on response filter out
+                  .onItem()
+                  .ifNotNull()
+                  .transformToMulti(
+                      response -> {
+                        List<String> authorizedKeyspace = new ArrayList<>(keyspaceNames);
+                        authorizedKeyspace.removeIf(
+                            keyspace -> {
+                              int index = keyspaceNames.indexOf(keyspace);
```

Comment:
Probably not significant since the list will be small most of the time, but this is a really bad way to do this. The inner function retraverses the list every time (O(n^2)) for no reason, we already know the index since the authorizations are in the same order as the names.

Compared to the v1 version:
```java
    List<T> filtered = new ArrayList<>(elements.size());
    for (int i = 0; i < elements.size(); i++) {
      if (authorizations.get(i)) {
        filtered.add(elements.get(i));
      }
    }
    return filtered;
```

Reply:
Adapting. Thx